OSX/Metabase/UI/MainViewController.m

```diff
@@ -133,8 +133,14 @@ - (void)saveCSV:(NSString *)apiURL {
 	if ([savePanel runModal] == NSFileHandlingPanelOKButton) {
 		NSLog(@"Will save CSV at: %@", savePanel.URL);
 		
-		NSURL *url = [NSURL URLWithString:apiURL relativeToURL:[NSURL URLWithString:BaseURL()]];
-		NSURLRequest *csvRequest = [NSURLRequest requestWithURL:url cachePolicy:NSURLRequestReloadIgnoringCacheData timeoutInterval:10.0f];
+		NSURL *url = [NSURL URLWithString:@"/api/dataset/csv" relativeToURL:[NSURL URLWithString:BaseURL()]];
+        NSString *postBody = [NSString stringWithFormat:@"query=%@",datasetQuery];
+        NSData *data = [postBody dataUsingEncoding:NSUTF8StringEncoding];
+		NSMutableURLRequest *csvRequest = [NSMutableURLRequest requestWithURL:url cachePolicy:NSURLRequestReloadIgnoringCacheData timeoutInterval:10.0f];
+        csvRequest.HTTPMethod = @"POST";
+        [csvRequest setValue:@"application/x-www-form-urlencoded" forHTTPHeaderField:@"Content-Type"];
+        [csvRequest setValue:[NSString stringWithFormat:@"%lu",(unsigned long)[data length]] forHTTPHeaderField:@"Content-Length"];
```

Comment:
A couple style points here:

*  put spaces after commas, just like in Java
*  `NSUInteger` is the canonical way to say unsigned long in Obj-C
*  just use dot syntax to get `data.length`

```objective-c
[NSString stringWithFormat:@"%lu", (NSUInteger)data.length]
```